File: src/AC/070/070.cpp
#include <vector>

class Solution {
public:
    int climbStairs(int n) {
        std::vector<int> stair_solution_n = {1, 2};
        stair_solution_n.resize(n);
        for (int i = 2; i < n; i++) {
            stair_solution_n[i] =
                stair_solution_n[i - 1] + stair_solution_n[i - 2];
        }

        return stair_solution_n[n - 1];
    }
};

File: src/AC/070/070.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        if n == 1 {
            return 1;
        }

        let mut dp = vec![0; n as usize];
        dp[0] = 1;
        dp[1] = 2;
        for i in 2..n as usize {
            dp[i] = dp[i - 1] + dp[i - 2];
        }

        dp[n as usize - 1]
    }
}
